src/opera/util/render_jinja2.py

```diff
@@ -202,6 +204,132 @@ def python_type_to_xml_type(obj) -> str:
+
+    Parameters
+    ----------
+    measured_parameters : dict
+       The GeoTIFF metadata from the output product. See get_geotiff_metadata()
+    mpc_path: str
+        Path to the Measured Parameters Descriptions YAML file
+    logger: PgeLogger
+        PgeLogger instance
+
+    Returns
+    -------
+    augmented_parameters : dict
+       The metadata fields converted to a list with name, value, types, etc
+    """
+    augmented_parameters = dict()
+
+    if mpc_path is not None:
+        with open(mpc_path) as f:
+            descriptions = yaml.safe_load(f)
+
+        missing_description_value = UNDEFINED_ERROR
+    else:
+        descriptions = dict()
+        missing_description_value = UNDEFINED_WARNING
+
+    for name, value in measured_parameters.items():
+        if isinstance(value, np.generic):
+            value = value.item()
+
+        if isinstance(value, np.ndarray):
+            value = value.tolist()
+
+        if isinstance(value, (list, dict)):
+            value = json.dumps(value, cls=NumpyEncoder)
+
+        guessed_data_type = python_type_to_xml_type(value)
+        guessed_attr_name = guess_attribute_display_name(name)
+
+        descriptions.setdefault(name, dict())
+
+        attr_description = descriptions[name].get('description', missing_description_value)
+        data_type = descriptions[name].get('attribute_data_type', guessed_data_type)
+        attr_type = descriptions[name].get('attribute_type', UNDEFINED_ERROR)
+        attr_name = descriptions[name].get('display_name', guessed_attr_name)
+        escape_html = descriptions[name].get('escape_html', False)
+
+        if escape_html:
+            value = html.escape(value)
+
+        augmented_parameters[name] = (
+            dict(name=attr_name, value=value, attr_type=attr_type,
+                 attr_description=attr_description, data_type=data_type)
+        )
+
+    return augmented_parameters
+
+
+def augment_hd5_measured_parameters(measured_parameters: dict, mpc_path: str, logger: PgeLogger) -> dict:
+    """
+    The augment_measured_parameters() function wrapped in a "preprocessing" step to
+    handle the structure of HDF5 metadata. While GeoTIFF metadata is a flat
+    dictionary, HDF5 metadata is a nested dictionary structure, wherein the variable
+    "keys" can be arbitrarily deep into the structure and the values likewise can be
+    nested dictionaries.
+
+    The preprocessing step in this method selectively flattens the metadata
+    dictionary based on the "paths" provided in the variable keys of the configuration
+    YAML file. The result of this preprocessing is then safely passed to the base
+    method to get the correct structure expected by the Jinja template.
+
+    Parameters
+    ----------
+    measured_parameters : dict
+        The HDF5 metadata from the output product. See get_cslc_s1_product_metadata()
+    mpc_path: str
+        Path to the Measured Parameters Descriptions YAML file
```

Comment:
I also think we should clarify that, unlike the baseline `augment_measured_parameters`, it is an error if `None` is provided or `mpc_path` does not exist.